client/engine/chainservice/mockchain.go

```diff
@@ -102,10 +107,10 @@ func (mc MockChain) handleTx(tx protocols.ChainTransaction) {
 		panic("unexpected chain transaction")
 	}
 
-	for _, out := range mc.out {
-		attemptSend(out, event)
-	}
-
+	mc.out.Range(func(key string, value chan Event) bool {
```

Comment:
nit: blank identifier makes the non-use explicit.

```suggestion
	mc.out.Range(func(_ string, value chan Event) bool {
```